core/src/main/scala/Block.scala

```diff
@@ -30,6 +30,14 @@ object Block {
       valueLength: Int = 90
   ): Block =
     Block.fromSource(Source.fromURI(path.toURI), keyLength, valueLength)
+
+  def sort(block: Block): Block = {
+    val sortedRecords = block.records.sortBy(_.key)
+    new Block(sortedRecords)
+  }
+  def sample(block: Block): List[Key] = {
+    block.records.map(_.key).toList
```

Comment:
In our design, `sample` method is in charge of sampling keys. E.g. not return *whole keys* of block.

Reply:
e.g. 
```scala
  def sample(by: Int = 10): LazyList[Key] = {
    def sampleFromRecords(
        records: LazyList[Record],
        by: Int = by
    ): LazyList[Key] = {
      val (current, rest) = records.splitAt(by)
      val head = current.head.key
      head #:: sampleFromRecords(rest, by)
    }

    sampleFromRecords(records, by)
  }
```

Reply:
Could I ask about the purpose of splitAt(interval)? It seems that splitAt(10) is intended to divide a list of Record objects into two parts at the specified index.
`Records -> (Records, Records)`
In the following line, we then obtain the key using the record.key.

From my perspective, splitAt(10)'s purpose is split Byte list to Key and dummy data like
`Byte -> (Key, dummy data)`
However, in this method, the argument is a list, not a byte

Reply:
I used `splitAt` to tie `n` keys and choose one. Other implementations welcomed. e.g. return `records.take(n)`
What's your idea?

Reply:
FYI) `records.take(n)` could be harmful. Note that we couldn't know length of lazy list only if all elements are calcutated. That means, if we choose `records.take(n)`, we should take records in a rate (e.g. `10%`), then calculate all elements to get length, then take 10% of length.

Reply:
I understand it! Thanks for the detailed explanation. There is an exception case that if the current is empty, we cannot bring it to the head of the current, so I added a base case for the recursion like
`    if (records.isEmpty)
      LazyList.empty[Key]`